### DISCUSSION TITLE
<html> element does not have a [lang] attribute

### DISCUSSION THREAD

**Discussion by Elindo586** [vercel/next.js#40169] [Help] [answered]
Well.... checking out lighthouse for errors, which I have a few but most everything is green...

One error found a lot is this one:

```
<html> element does not have a [lang] attribute

If a page doesn't specify a lang attribute, a screen reader assumes that the page is in the default language that the user chose when setting up the screen reader. If the page isn't actually in the default language, then the screen reader might not announce the page's text correctly. [Learn more](https://web.dev/html-has-lang/?utm_source=lighthouse&utm_medium=devtools).

```

![Capture88](https://user-images.githubusercontent.com/93617433/188031628-46c881f2-efeb-472a-b435-98fa222c1fbc.PNG)

Granted I haven't used the lang tag much, but trying to fix one page I got on some of the main div s to see if it would fix it, and the error still showing there...

I wonder... is lighthouse asking me to put it the html tag????    well.....   Where in the world do I find an html tag????   plus...  not everything would be in English, but I also have some content written in Spanish inside a page that is using  the main Nav that are in English.... 

What would be a recommended way to fix this thing???

**Comment by icyJoseph** [comment, marked as answer]
Create a page called, `_document.js` or `_document.tsx`:

```js
import { Html, Head, Main, NextScript } from 'next/document'

export default function Document() {
  return (
    <Html lang="en">
      <Head />
      <body>
        <Main />
        <NextScript />
      </body>
    </Html>
  )
}
```

Now to the point of multi-language... I personally think that you should just use the lang attribute for the sections that use a different language, refer to https://developer.mozilla.org/en-US/docs/Web/HTML/Global_attributes/lang:

```html
<p>This paragraph is English, but the language is not specifically defined.</p>

<p lang="en-GB">This paragraph is defined as British English.</p>

<p lang="es">Este párrafo es definido en Español.</p>
```

I think one has to be extremely careful with this things because they affect accessibility (a11y). In this case I think it should be kind of safe to specify the language for a section of the page, at the end of the day it's just two languages.

This is not a Next.js thing itself, it is proper to web dev in general, I think if you really want to get to the roots of its, there's tons to read in a11y websites and StackOverflow.

**Reply by Elindo586** [reply]
thanks.... after all that _document.js was needed for something...
